querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/HibernateQueryFactory.java

```diff
@@ -37,6 +37,20 @@
 
     private final Provider<Session> session;
 
+    public HibernateQueryFactory(Session session) {
+        this(HQLTemplates.DEFAULT, session);
+    }
+
+    public HibernateQueryFactory(JPQLTemplates templates, final Session session) {
+        this.session = new Provider<Session>() {
+            @Override
+            public Session get() {
+                return session;
+            }
+        };
+        this.templates = templates;
```

Comment:
You could also create the `Provider<Session>` and pass it to the existing constructor `this(templates, new Provider<Session>...);`.